### PR TITLE
fix: reduce project overview query count to 2.

### PR DESCRIPTION
**Pull request by chriswk**
Previously we've been doing N+1 queries for projects.

So until we've decided what to keep of the numbers. This should at least scale better than our current solution

### CODE REVIEW

**Comment by chriswk** [comment]
[Loading projects is way to slow with many projects.](https://trello.com/c/NnCAMOOh/875-loading-projects-is-way-to-slow-with-many-projects)

File: src/lib/db/project-store.ts [hunk 1]
```diff
@@ -44,13 +54,62 @@ class ProjectStore implements IProjectStore {
...
+            this.mapProjectWithCountRow,
+        );
+        projectTimer();
+        const memberTimer = this.timer('getMemberCount');
+        const memberCount = await this.db.raw(
+            `SELECT count(role_id) as member_count, project FROM role_user GROUP BY project`,
+        );
+        memberTimer();
+        const memberMap = memberCount.rows.reduce((a, r) => {
+            a[r.project] = parseInt(r.member_count, 10);
```
Comment (sighphyre) [inline src/lib/db/project-store.ts:92]: Do we need the radix here? The two cases I know off the top of my head are on old JS interpreters with being too eager to interpret this as hex and passing parseInt to be invoked by a map/reduce/filter. I don't think we have either case here.

But do we want it for explicit's sake?

**Review by sighphyre — APPROVED** [review]
Looks good, nice change here, I can imagine this is going to be much faster  and use less number crunching. Should we think about some performance testing to measure this kind of thing?

File: src/lib/db/project-store.ts (hunk 1 above)
Comment (chriswk) [reply src/lib/db/project-store.ts:92]: Usually knex have handled this when typing the rows already, so it shouldn't be necessary, but I had tests failiing, this solved it, and I like to be explicit with radixes when using parseInt anyway, just so nobody thinks I've got a funky idea about base 8 :)

Comment (chriswk) [reply src/lib/db/project-store.ts:92]: Let me have a go at writing it in knex syntax instead to see if that is better

File: src/lib/db/project-store.ts [hunk 2]
```diff
@@ -44,13 +54,62 @@ class ProjectStore implements IProjectStore {
...
+        let projectsQuery = this.db(TABLE)
+            .select(
+                this.db.raw(
+                    'projects.id, projects.name, projects.description, projects.health, projects.updated_at, count(features.name) AS number_of_features',
+                ),
+            )
+            .leftJoin('features', 'features.project', 'projects.id')
+            .groupBy('projects.id');
+        if (query) {
+            projectsQuery = projectsQuery.where(query);
```
Comment (ivarconr) [inline src/lib/db/project-store.ts:77]: Low: a bit confusing naming here `projectsQuery` vs `query`.

File: src/lib/db/project-store.ts (hunk 1 above)
Comment (chriswk) [reply src/lib/db/project-store.ts:92]: Ah, Note that in Postgres, count returns a bigint type which will be a String and not a Number, that explains why it became a string. Fair enough, then we'll still need this parsing. We'll have performance problems way before we hit javascript accuracy issues due to reducing a bigint to int.

File: src/lib/db/project-store.ts [hunk 3]
```diff
@@ -208,5 +267,6 @@ class ProjectStore implements IProjectStore {
         };
     }
 }
+
 export default ProjectStore;
 module.exports = ProjectStore;
```
Comment (ivarconr) [inline src/lib/db/project-store.ts:272]: should also be removed as we only do ts now.

**Review by ivarconr — APPROVED** [review]

File: src/lib/db/project-store.ts (hunk 1 above)
Comment (chriswk) [reply src/lib/db/project-store.ts:92]: Maybe better to just use the `+` in front?

Comment (chriswk) [reply src/lib/db/project-store.ts:92]: Thanks @ivarconr that seems better :)

File: src/lib/db/project-store.ts (hunk 2 above)
Comment (chriswk) [reply src/lib/db/project-store.ts:77]: Indeed, I'll change

**Pull request merged**